mabwiser/popularity.py

```diff
@@ -33,9 +33,17 @@ def partial_fit(self, decisions: np.ndarray, rewards: np.ndarray, contexts: np.n
 
     def predict(self, contexts: np.ndarray = None) -> Arm:
 
-        # Select an arm randomized by expectation probability
+        # Return the first arm with maximum expectation
+        return argmax(self.predict_expectations())
+
+    def predict_expectations(self, contexts: np.ndarray = None) -> Dict[Arm, Num]:
+
+        # Return a random value between 0 and 1 for each arm that is "proportional" to the
+        # expectation of the arm, by sampling from a Dirichlet distribution
```

Comment:
Quick question: have we used Dirichlet for sampling before or is it part of the re-factor? I may not be as close to the mabwiser code base but this is the first time I am seeing it used.

Reply:
It is the first time we're using in MABWiser.

Reply:
Can you say a few words on why is popularity a Dirichlet distro and how is this diff than the multivariate dist we have already? Dir

Reply:
For Popularity we calculate the average reward (assuming positive rewards) and then normalize it such that it sums to 1. We arm A has a value of 0.1 and arm B has a value of 0.2 (i..e 2x more popular), we want to predict B 2x more often than A. 

In the current version we achieve this using the [choice](https://numpy.org/doc/stable/reference/random/generated/numpy.random.choice.html) function, which randomly selects an arm according to the normalized values (weights). The Dirichlet distribution can be seen as a multivariate generalization of the Beta distribution and it can be used to sample the normalized values itself. I.e., for each arm it samples a random value between 0 and 1 in proportion to normalized values such that it sums to 1.

The multivariable normal distribution we have already assumes a normal distribution with a given mean and covariance. Values can be positive and negative and has no constraint wrt to the sum of the values.

Reply:
Ah thanks! I was overlooking the "sum" detail

Reply:
Btw, it might help to incorporate some your comment here to the code comments about Dirichlet